.github/workflows/commit_syntax.yml

```diff
@@ -0,0 +1,32 @@
+name: 'Commit Message Check'
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - '*'
+
+jobs:
+  check-commit-message:
+    name: Check Commit Message
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Commit Type
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^(Breaking|Update|Fix|New|Upgrade|No code):'
+          flags: ''
+          error: 'Your commit message should start with a valid semantic release verb as defined in package.json. You may also prepend "No code:" to indicate the commit does not contain changes to the released code.'
+          # ideally we should check all commit messages, but we've yet to create a github token to do so.
```

Comment:
Shouldn't the built-in `GITHUB_TOKEN` work here? See: https://docs.github.com/en/actions/reference/authentication-in-a-workflow